feat: added snappy compression support

m0lc14kk opened the pull request:
Added support for `snappy` compression algorithm.

Compression is provided by `snappy` package, which is lightweight and the fastest way to use this type of compression.

oliverperzyk committed:
feat: added snappy compression support

oliverperzyk committed:
fix: fixed lint errors

extremeheat commented:
This is alright, but the problem is that it's a C++ library and we'd like to move away from requiring any native dependencies at all to avoid needing to install a bunch of native build tools and other complexity that comes along with it. A pure JS or WASM solution would be more ideal, but we can offer a native option too ideally just not a required dep in package.json (like how we support different raknet backends).

m0lc14kk commented:
okay, so if I would use a pure JS snappy solution and use it in bedrock-protocol, you would be okay with it and applied it to the next version?

I just found [snappyjs](https://github.com/zhipeng-jia/snappyjs/blob/master/dist/snappyjs.min.js) minified file, which bedrock-protocol can use to handle snappy compressions

what do you think about copying this minified file or maybe just using this dependency?

oliverperzyk committed:
feat: moved from snappy to snappyjs

Changed dependency from snappy to snappyjs, to have a pure implementation of an algorithm.

oliverperzyk committed:
feat: added a test for snappy compression

Added a Snappy example in tests.

m0lc14kk commented:
I just added a small test and changed the library, you can tell me if that's enough to add this to bedrock-protocol.

oliverperzyk committed:
fix: fixed lint errors

oliverperzyk committed:
fix: changed quotes to match lint standards

extremeheat commented:
CI is failing

And for the test you can update one of the internal server tests to set snappy as the compressor
